pkg/monitor/monitorapi/identification_pod.go

```diff
@@ -0,0 +1,176 @@
+package monitorapi
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+func LocatePod(pod *corev1.Pod) string {
+	return fmt.Sprintf("ns/%s pod/%s node/%s uid/%s", pod.Namespace, pod.Name, pod.Spec.NodeName, pod.UID)
+}
+
+func LocatePodContainer(pod *corev1.Pod, containerName string) string {
+	return fmt.Sprintf("ns/%s pod/%s node/%s uid/%s container/%s", pod.Namespace, pod.Name, pod.Spec.NodeName, pod.UID, containerName)
+}
+
+func PodFrom(locator string) PodReference {
+	parts := LocatorParts(locator)
+	namespace := NamespaceFrom(parts)
+	name := parts["pod"]
+	uid := parts["uid"]
+	if len(namespace) == 0 || len(name) == 0 || len(uid) == 0 {
+		return PodReference{}
+	}
+	return PodReference{
+		NamespacedReference: NamespacedReference{
+			Namespace: namespace,
+			Name:      name,
+			UID:       uid,
+		},
+	}
+}
+
+func ContainerFrom(locator string) ContainerReference {
+	pod := PodFrom(locator)
+	parts := LocatorParts(locator)
+	name := parts["container"]
+	if len(name) == 0 || len(pod.UID) == 0 {
+		return ContainerReference{}
+	}
+	return ContainerReference{
+		Pod:           pod,
+		ContainerName: name,
+	}
+}
+
+type PodReference struct {
+	NamespacedReference
+}
+
+func (r PodReference) ToLocator() string {
+	return fmt.Sprintf("ns/%s pod/%s uid/%s", r.Namespace, r.Name, r.UID)
+}
+
+type ContainerReference struct {
+	Pod           PodReference
+	ContainerName string
+}
+
+func (r ContainerReference) ToLocator() string {
+	return fmt.Sprintf("ns/%s pod/%s uid/%s container/%s", r.Pod.Namespace, r.Pod.Name, r.Pod.UID, r.ContainerName)
+}
+
+func ReasonFrom(message string) string {
+	tokens := strings.Split(message, " ")
+	annotations := map[string]string{}
+	for _, curr := range tokens {
+		if !strings.Contains(curr, "/") {
+			continue
+		}
+		annotationTokens := strings.Split(curr, "/")
+		annotations[annotationTokens[0]] = annotationTokens[1]
```

Comment:
> small savings but can we just do this:

if annotationTokens[0] == "reason" {
  return annotationTokens[1]
}
though I'm not sure what to return if you didn't find "reason" in the tokens.

I'd like to keep the logic that produces the annotations because I suspect we will need it again.